extra_data/sourcedata.py

```diff
@@ -15,6 +15,8 @@ class SourceData:
 
     Don't create this directly; get it from ``run[source]``.
     """
+    _first_source_file = ...
```

Comment:
Welcome to team `...` :grinning: 

I tend to only do this if `None` is a possible value in the cache. But I've no objection to using `...` even if None would work.

Reply:
In general I agree, but I did this in preparation and to be symmetric with `_device_class` 😉